website/docs/ref/core.md

```diff
@@ -115,16 +115,32 @@ Formatting of messages as strings (e.g: `"My name is {name}"`) works in developm
 The same example would in real application look like this:
 
 ```ts
-import { i18n } from "@lingui/core"
+import { i18n } from "@lingui/core";
 
 // File generated by `lingui compile`
-import { messages: messagesEn } from "./locale/en/messages.js"
+import { messages as messagesEn } from "./locale/en/messages.js";
 
-i18n.load('en', messagesEn)
+i18n.load("en", messagesEn);
 ```
 
 :::
 
+### `i18n.setMessagesCompiler(compiler)` {#i18n.setMessagesCompiler}
+
+Registers a `MessageCompiler` to enable the use of uncompiled catalogs at runtime.
+
+In production builds, the `MessageCompiler` is typically excluded to reduce bundle size.
+
+By default, message catalogs should be precompiled during the build process. However, if you need to compile catalogs at runtime, you can use this method to set a message compiler.
```

Comment:
As a user reading the docs, I understand what the function does (the `setMessagesCompiler` is also quite self-explanatory) but I don't really understand why I'd need such feature in the first place. Could we have a paragraph about what problem it solves and how to use it? 

I know writing docs is annoying but I think it'd be great to not wait until a later epic is finished.
The PR mentions "the proper macro config" - it'd be great to know what that is, otherwise that can be quite frustrating for consumers since @timofei-iatsenko is probably the only one who knows that "proper config". 🙈 
Apologies and thanks :)

Reply:
There is still functionality needed to be done before this article could be written. I don't want to bring too much user attention to that while it would be 100% ready.  That was also the reason why I didn't write this doc in the first place. 

This is a breaking change, due to message catalog format changes, and i wanted this to be included in next major release. The rest features needed to make this work would not be breaking and could be done later.